addons/binding/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/ApplianceGroheOndusHandler.java

```diff
@@ -0,0 +1,256 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.grohe.ondus.api.OndusService;
+import org.grohe.ondus.api.model.Appliance;
+import org.grohe.ondus.api.model.ApplianceCommand;
+import org.grohe.ondus.api.model.ApplianceData;
+import org.grohe.ondus.api.model.ApplianceData.Measurement;
+import org.grohe.ondus.api.model.Location;
+import org.grohe.ondus.api.model.Room;
+import org.openhab.binding.groheondus.internal.ApplianceGroheOndusConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Florian Schmidt and Arne Wohlert - Initial contribution
+ */
+@NonNullByDefault
+public class ApplianceGroheOndusHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ApplianceGroheOndusHandler.class);
+
+    @Nullable
+    private ApplianceGroheOndusConfiguration config;
+
+    public ApplianceGroheOndusHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(ApplianceGroheOndusConfiguration.class);
+        scheduler.submit(() -> {
+            updateChannels();
+            updateStatus(ThingStatus.ONLINE);
+        });
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        super.channelLinked(channelUID);
+
+        OndusService ondusService = getOndusService();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return;
+        }
+        Appliance appliance = getAppliance(ondusService);
+        if (appliance == null) {
+            return;
+        }
+        updateChannel(channelUID, appliance, getLastMeasurement(appliance));
+    }
+
+    public void updateChannels() {
+        OndusService ondusService = getOndusService();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return;
+        }
+        Appliance appliance = getAppliance(ondusService);
+        if (appliance == null) {
+            return;
+        }
+
+        Measurement measurement = getLastMeasurement(appliance);
+        getThing().getChannels().forEach(channel -> updateChannel(channel.getUID(), appliance, measurement));
+
+        int pollingInterval = getPollingInterval(appliance);
+        scheduler.schedule(this::updateChannels, pollingInterval, TimeUnit.SECONDS);
+    }
+
+    private int getPollingInterval(Appliance appliance) {
+        if (config.pollingInterval > 0) {
+            return config.pollingInterval;
+        }
+        return appliance.getConfig().getMeasurementTransmissionIntervall();
```

Comment:
`getMeasurementTransmissionIntervall()` has a typo because it ends with two L's. Can you fix that in the API?

If the API requires Java 8  you might also want to return something like a `java.util.Duration` here so people don't have to second guess the unit.

Reply:
Unfortunately, the GROHE API returns the following JSON property:
`measurement_transmission_intervall`

The method just represents that property. It's already "misspelled" there, so I don't feel confident changing the name of the getter method, as this would confuse people, I think :/

Regarding your suggestion with returning a Java 8 Duration type: This sounds like a good idea for me, however, it would change the data type from just representing what was send using the API to include some logic in it, as JSOn does not know the data type by itself. I would like to have this discussed as an issue in https://github.com/FlorianSW/grohe-ondus-api-java first, and maybe think about it a bit more before changing it, I hope that's ok for you? :)

Reply:
It's OK with me to skip this one for now.